deno/lib/types.ts

```diff
@@ -3488,6 +3488,175 @@ export class ZodNaN extends ZodType<number, ZodNaNDef> {
   };
 }
 
+/////////////////////////////////////////
+/////////////////////////////////////////
+//////////                     //////////
+//////////      ZodFile         //////////
+//////////                     //////////
+/////////////////////////////////////////
+/////////////////////////////////////////
+
+type ZodFileCheck =
+  | { kind: "max"; value: number; message?: string }
+  | { kind: "min"; value: number; message?: string }
+  | { kind: "type"; value: string | string[]; message?: string };
+
+export interface ZodFileDef extends ZodTypeDef {
+  typeName: ZodFirstPartyTypeKind.ZodFile;
+  checks: ZodFileCheck[];
+}
+
+export class ZodFile<FileConstructor extends Function> extends ZodType<
+  FileConstructor,
```

Comment:
Would be possible to support an Array of `File`s?

The use case is if people use [multiple=true](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-multiple) in an `<input type="file" multiple />`

This way these upload can be processed in the server and use this validation like this:

```
const myRequest = z.object({
  gallery: z.file({}, File).array()
})
```

But for this to work we need to manage the case when the input is an array. 

I would play with it in my dev environment with your fork : )

Reply:
Forget it. This works:
``` javascript
const myRequest = z.object({
  gallery: z.array(z.file({}, File))
})
```

Nothing to do here : )